component/azstorage/block_blob.go

```diff
@@ -78,6 +79,8 @@ type BlockBlob struct {
 	blockLocks      common.KeyedMutex
 }
 
+type Metadata map[string]*string
```

Comment:
We are using this map in ObjAttr, correct place to define this data type would be inside in the common where ObjAttr is defined.